database/sqlite/sqlite_context.c

```diff
@@ -0,0 +1,597 @@
+    }
+
+    rc = sqlite3_bind_int(res, 10, (time_t) context_data->deleted);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind last_time_t to store context details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_text(res, 11, context_data->family, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context to store details");
+        goto failed;
+    }
+
+    rc_stored = execute_insert(res);
+
+    if (rc_stored != SQLITE_DONE)
+        error_report("Failed store context details for context %s, rc = %d", context_data->id, rc_stored);
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement that stores context details, rc = %d", rc);
+
+    return (rc_stored != SQLITE_DONE);
+}
+
+// Delete a context
+
+#define CTX_DELETE_CONTEXT "DELETE FROM context WHERE host_id = @host_id AND id = @context;"
+int ctx_delete_context(uuid_t *host_uuid, VERSIONED_CONTEXT_DATA *context_data)
+{
+    int rc, rc_stored = 1;
+    sqlite3_stmt *res = NULL;
+
+    if (unlikely(!context_data || !context_data->id))
+        return 0;
+
+
+    rc = sqlite3_prepare_v2(db_context_meta, CTX_DELETE_CONTEXT, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to delete context");
+        return 1;
+    }
+
+	rc = sqlite3_bind_blob(res, 1, *host_uuid, sizeof(*host_uuid), SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind host_id to delete context data");
+        goto failed;
+    }
+
+
+    rc = sqlite3_bind_text(res, 2, context_data->id, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context id for data deletion");
+        goto failed;
+    }
+
+    rc_stored = execute_insert(res);
+
+    if (rc_stored != SQLITE_DONE)
+        error_report("Failed to delete context %s, rc = %d", context_data->id, rc_stored);
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement where deleting a context, rc = %d", rc);
+
+    return (rc_stored != SQLITE_DONE);
+}
+
+//
+// TESTING FUNCTIONS
+//
+static void dict_ctx_get_label_list_cb(SQL_CLABEL_DATA *label_data_ptr, void *data)
+{
+    (void)data;
+    SQL_CLABEL_DATA *label_data = label_data_ptr;
+
+    info(" LABEL %d %s = %s", label_data->label_source, label_data->label_key, label_data->label_value);
+}
+
+static void dict_ctx_get_dimension_list_cb(SQL_DIMENSION_DATA *dimension_data_ptr, void *data)
+{
+    (void)data;
+
+    SQL_DIMENSION_DATA *dimension_data = dimension_data_ptr;
+
+    char uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower(dimension_data->dim_id, uuid_str);
+
+    info(" Dimension %s = %s", uuid_str, dimension_data->id);
+}
+
+
+static void dict_ctx_get_chart_list_cb(SQL_CHART_DATA *chart_data, void *data)
+{
+    (void)data;
+
+    char uuid_str[GUID_LEN + 1];
```

Comment:
Please, change from `GUID_LEN + 1` to `UUID_LEN`.  The code is not wrong, but considering we are talking about `uuid` here, it will make more sense.

Reply:
Replaced with UUID_STR_LEN